### PR TITLE
allow setting run mode via env, add helpers to determine it

### PR DESCRIPTION
**Pull request by benedikt-bartscher**
bonus: error out if backend-only and frontend-only is set at the same time

needed here: https://github.com/reflex-dev/reflex-chakra/pull/22

### CODE REVIEW

File: reflex/utils/exec.py [hunk 1]
```diff
@@ -496,6 +496,24 @@ def is_prod_mode() -> bool:
     return current_mode == constants.Env.PROD.value
 
 
+def is_frontend_only() -> bool:
+    """Check if the app is running in frontend-only mode.
+
+    Returns:
+        True if the app is running in frontend-only mode.
+    """
+    return os.environ.get(constants.ENV_FRONTEND_ONLY_ENV_VAR, "").lower() == "true"
```
[the remark this answers is not in the document]
Comment (benedikt-bartscher) [reply reflex/utils/exec.py:505]: i know, its a mess currently. i think this needs a refactor over the whole repo, maybe we should address that one in a different PR later on.

[the remark this answers is not in the document]
Comment (benedikt-bartscher) [reply reflex/utils/exec.py:505]: i created https://github.com/reflex-dev/reflex/issues/4172 to discuss/refactor boolean env var handling

[the remark this answers is not in the document]
Comment (Lendemor) [reply reflex/utils/exec.py:505]: Thanks for making that issue.

Yeah I'll approve this PR and we can clean up things over the whole repo in a different PR.

**Review by Lendemor — APPROVED** [review]

**Comment by benedikt-bartscher** [comment]
needs coordination with https://github.com/reflex-dev/reflex/pull/4192

**Pull request merged**